packages/yew-macro/src/function_component.rs

```diff
@@ -171,7 +175,12 @@ pub fn function_component_impl(
         name: function_name,
         return_type,
     } = component;
-
+    let component_name = component_name.unwrap_or_else(|| function_name.clone());
+    let function_name = format_ident!(
+        "{}FunctionProvider",
+        function_name,
+        span = function_name.span()
+    );
```

Comment:
do you want to convert the function name to PascalCase with [convert_case](https://docs.rs/convert_case/0.4.0/convert_case/enum.Case.html)?

Reply:
It doesn't really matter since it's not in the public API. Besides, generally this name would be PascalCase anyway because that's the convention for component names.